llama_index/vector_stores/pinecone.py

```diff
@@ -255,12 +260,17 @@ def add(
                 VECTOR_KEY: node.get_embedding(),
                 METADATA_KEY: metadata,
             }
-            if self.add_sparse_vector and self._tokenizer is not None:
-                sparse_vector = generate_sparse_vectors(
-                    [node.get_content(metadata_mode=MetadataMode.EMBED)],
-                    self._tokenizer,
-                )[0]
-                entry[SPARSE_VECTOR_KEY] = sparse_vector
+            if self.add_sparse_vector:
```

Comment:
could we add sparse vector this option to base `BasePydanticVectorStore` abstraction instead of only for Pinecone.
is only Pinecone supporting add spare vector?
thanks!

Reply:
Yeah that seems reasonable. Looks like [https://weaviate.io/developers/weaviate/api/graphql/search-operators#hybrid](Weaviate) supports sparse vectors as well.

Reply:
thanks could you help do the refactor a bit